text/0258-env-context-baggage-carriers.md

```diff
@@ -0,0 +1,294 @@
+# Environment Variable Specification for Context and Baggage Propagation
+
+This is a proposal to add Environment Variables to the OpenTelemetry
+specification as a carrier for context and baggage propagation between
+processes.
+
+## Table of Contents
+
+* [Motivation](#motivation)
+* [Design](#design)
+  * [Example Context](#example-context)
+    * [Distributed Tracing in OpenTofu Prototype Example](#distributed-tracing-in-opentofu-prototype-example)
+* [Core Specification Changes](#core-specification-changes)
+  * [UNIX](#unix-limitations)
+  * [Windows](#windows-limitations)
+  * [Allowed Characters](#allowed-characters)
+* [Trade-offs and Mitigations](#trade-offs-and-mitigations)
+  * [Case-sensitivity](#case-sensitivity)
+  * [Security](#security)
+* [Prior Art and Alternatives](#prior-art-and-alternatives)
+  * [Alternatives and why they were not chosen](#alternatives-and-why-they-were-not-chosen)
+* [Open Questions](#open-questions)
+* [Future Possibilities](#future-possibilities)
+
+## Motivation
+
+The motivation for defining the specification for context and baggage
+propagation by using environment variables as carriers stems from a long open
+issue on the OpenTelemetry Specification repository, [issue #740][issue-740].
+This issue has been open for such a long time that multiple groups have gone
+forward in implementing their own solutions to the problem using `TRACEPARENT`
+and `TRACESTATE` environment variables.
+
+[Issue #740][issue-740] identifies several use cases of systems that do not
+communicate across bounds by leveraging HTTP communications such as:
```

Comment:
```suggestion
[Issue #740][issue-740] identifies several use cases of systems that do not
communicate across bounds by leveraging network communications such as:
```

I guess HTTP is fine, but my understanding is that the use cases below are process-boundaries vs network-boundaries which have headers, meta data and other existing mechanisms to transfer context.